src/neo/Network/P2P/Payloads/OracleResponseSignature.cs

```diff
@@ -0,0 +1,111 @@
+using Neo.Cryptography;
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Neo.Network.P2P.Payloads
+{
+    public class OracleResponseSignature : ISerializable
+    {
+        private const byte ResponseSignatureType = 0x01;
+
+        private UInt256 _transactionRequestHash;
+        public UInt256 TransactionRequestHash
+        {
+            get => _transactionRequestHash;
+            set { _transactionRequestHash = value; _hash = null; _size = 0; }
+        }
+
+        private UInt256 _oracleExecutionCacheHash;
+        public UInt256 OracleExecutionCacheHash
+        {
+            get => _oracleExecutionCacheHash;
+            set { _oracleExecutionCacheHash = value; _hash = null; _size = 0; }
+        }
+
+        /// <summary>
+        /// Signature for the oracle response tx for this public key
+        /// </summary>
+        private byte[] _signature;
+        public byte[] Signature
+        {
+            get => _signature;
+            set
+            {
+                if (value.Length != 64) throw new ArgumentException();
+                _signature = value;
+                _hash = null;
+                _size = 0;
+            }
+        }
+
+        private int _size;
+        public int Size
+        {
+            get
+            {
+                if (_size == 0)
+                {
+                    _size = sizeof(byte) +      //Type
+                        UInt256.Length +        //Transaction Hash
+                        UInt256.Length +        //OracleExecutionCache Hash
+                        Signature.GetVarSize(); //Oracle Validator Signature
```

Comment:
If it's a fixed size it must be a fixed length 

```suggestion
                        Signature.Length; //Oracle Validator Signature
```